libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -269,6 +269,24 @@ bool SyntaxChecker::visit(VariableDeclaration const& _declaration)
 	return true;
 }
 
+bool SyntaxChecker::visit(VariableDeclarationStatement const& _statement)
+{
+	// Check if declaration contains at least one named component, and report otherwise.
+	size_t const namedVariablesCount = count_if(
+		_statement.declarations().begin(),
+		_statement.declarations().end(),
+		[](ASTPointer<VariableDeclaration> const& decl) { return decl.get() != nullptr; }
+	);
+
+	if (namedVariablesCount == 0)
+		m_errorReporter.syntaxError(
+			_statement.location(),
+			"Variable declaration statement does not declare any variables."
```

Comment:
This can only happen with ``var``, so I would mention the removal of ``var`` in the error message and suggest to remove the superfluous declaration altogether, since it doesn't declare anything anyways.